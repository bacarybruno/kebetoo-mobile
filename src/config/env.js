/* eslint-disable no-undef */

__DEV__ = false

// TODO: use env variables
export default {
  apiBaseUrl: __DEV__ ? 'http://localhost:1337' : 'https://kebetoo.herokuapp.com',
  assetsBaseUrl: __DEV__ ? 'http://localhost:1337' : 'https://kebetoo.s3.eu-west-3.amazonaws.com',
}
