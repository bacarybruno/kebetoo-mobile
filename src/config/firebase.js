import { utils } from '@react-native-firebase/app';
import analytics from '@react-native-firebase/analytics';

if (utils().isRunningInTestLab) {
  // do not track analytics for firebase test lab
  // used by google play to run test on publish
  analytics().setAnalyticsCollectionEnabled(false);
}
