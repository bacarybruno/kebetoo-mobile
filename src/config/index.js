export { default as env } from './env';
export { default as strings } from './strings';
