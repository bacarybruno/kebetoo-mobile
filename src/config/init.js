import { GoogleSignin } from '@react-native-community/google-signin'
import dayjs from 'dayjs'
import relativeTime from 'dayjs/plugin/relativeTime'

GoogleSignin.configure({
  scopes: ['email', 'profile', 'openid'],
  webClientId: '355888072499-spu5d874qrfb6tfs3pq195cjf6ij56go.apps.googleusercontent.com',
})

dayjs.extend(relativeTime)
