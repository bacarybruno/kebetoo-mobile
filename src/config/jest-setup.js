/* eslint-disable import/no-extraneous-dependencies */
import { NativeModules } from 'react-native';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MockDate from 'mockdate';
import configureStore from 'redux-mock-store';
import 'react-native-gesture-handler/jestSetup';
import { renderHook } from '@testing-library/react-hooks';
import auth from '@react-native-firebase/auth';
import { SafeAreaContext } from '@app/shared/contexts';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { PortalProvider } from '@gorhom/portal';

// Silence the warning https://github.com/facebook/react-native/issues/11094#issuecomment-263240420
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

// wix/react-native-keyboard-input
NativeModules.KeyboardTrackingViewTempManager = {
  KeyboardTrackingScrollBehaviorNone: 'NONE',
  KeyboardTrackingScrollBehaviorScrollToBottomInvertedOnly: 'SCROLL_TO_BOTTOM_INVERTED_ONLY',
  KeyboardTrackingScrollBehaviorFixedOffset: 'FIXED_OFFSET',
};

NativeModules.RealPathUtils = {
  getOriginalFilePath: jest.fn(),
};

const mockedStoreState = {
  postsReducer: {
    posts: [],
    authors: [],
  },
  notificationsReducer: {
    notifications: [],
  },
  userReducer: {
    profile: {
      ...auth().currentUser,
      isLoggedIn: true,
    },
  },
  appReducer: {
    theme: 'system',
    locale: 'fr',
  },
};
// helper to setup unit tests
/* eslint-disable comma-dangle */
const setupTest = (WrappedComponent, renderFn = TestRenderer.create) => {
  let wrapper = null;
  return (defaultProps = {}) => (additionalProps = {}) => {
    const {
      store, __storeState__ = mockedStoreState, disconnectNavigation, ...props
    } = defaultProps;
    const propsWithArgs = {
      ...props,
      ...additionalProps,
    };
    const mockStore = configureStore();
    const safeAreaCtxValue = {
      updateTopSafeAreaColor: jest.fn(),
      updateBottomSafeAreaColor: jest.fn(),
      resetStatusBars: jest.fn(),
    };
    const Stack = createStackNavigator();
    const Component = (
      <SafeAreaContext.Provider value={safeAreaCtxValue}>
        <Provider store={store || mockStore(__storeState__)}>
          <PortalProvider>
            {disconnectNavigation
              ? <WrappedComponent {...propsWithArgs} />
              : (
                <NavigationContainer>
                  <Stack.Navigator>
                    <Stack.Screen name="jest">
                      {() => <WrappedComponent {...propsWithArgs} />}
                    </Stack.Screen>
                  </Stack.Navigator>
                </NavigationContainer>
              )}
          </PortalProvider>
        </Provider>
      </SafeAreaContext.Provider>
    );
    wrapper = renderFn(Component);
    return { wrapper, props: propsWithArgs };
  };
};

export const setupHook = (useHook, ...props) => {
  const mockStore = configureStore();
  const store = mockStore(mockedStoreState);
  const wrapper = ({ children }) => (
    <Provider store={store}>
      {children}
    </Provider>
  );
  const rendered = renderHook(() => useHook(...props), {
    wrapper,
  });
  return rendered;
};

/**
 * Explicit mocks
 * Create mocks here only if they can't be created on the __mocks__ folder
 */

jest.useFakeTimers();
jest.setTimeout(30000);

// react-navigation
const mockedNavigate = jest.fn();
const mockedSetOptions = jest.fn();
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    navigate: mockedNavigate,
    setOptions: mockedSetOptions,
    addListener: jest.fn(),
  }),
  useRoute: () => ({
    params: {},
  }),
  useFocusEffect: (cb) => cb()
}));
jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock');

  // The mock for `call` immediately calls the callback which is incorrect
  // So we override it with a no-op
  Reanimated.default.call = () => { };
  Reanimated.default.addWhitelistedUIProps = () => { };

  return Reanimated;
});

// react-native-screens
jest.mock('react-native-screens', () => {
  const Screens = jest.requireActual('react-native-screens');
  Screens.enableScreens = () => { };
  return Screens;
});

// date
MockDate.set(new Date(Date.parse('2020-06-25T13:30:00+02:00')));

export default setupTest;
