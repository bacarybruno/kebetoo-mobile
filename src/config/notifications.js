import { Platform } from 'react-native'
import PushNotification from 'react-native-push-notification'
import AsyncStorage from '@react-native-community/async-storage'
import messaging from '@react-native-firebase/messaging'

import { getNotificationMessage, getNotificationTitle } from '@app/features/notifications/containers'

PushNotification.createChannel({
  channelId: 'kbt',
  channelName: 'Kebetoo Channel',
  channelDescription: 'Local notifications sent by Kebetoo',
  playSound: false,
})

const formatNotificationMessage = ({ name, message }) => `${name} ${message}`

PushNotification.configure({
  requestPermissions: Platform.OS === 'ios',
})

// handle background notifications
messaging().setBackgroundMessageHandler(async (remoteMessage) => {
  const bgNotifications = JSON.parse(await AsyncStorage.getItem('backgroundNotifications')) || []
  bgNotifications.push(remoteMessage)
  PushNotification.setApplicationIconBadgeNumber(bgNotifications.length)
  await AsyncStorage.setItem('backgroundNotifications', JSON.stringify(bgNotifications))

  const notificationTitle = getNotificationTitle(remoteMessage)
  if (!notificationTitle && !remoteMessage.data.title) return

  PushNotification.localNotification({
    channelId: 'kbt',
    title: notificationTitle
      ? formatNotificationMessage(notificationTitle)
      : remoteMessage.data.title,
    message: getNotificationMessage(remoteMessage) || remoteMessage.data.message,
    smallIcon: 'ic_stat_ic_notification',
  })
})
