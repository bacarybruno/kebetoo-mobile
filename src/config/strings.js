import LocalizedStrings from 'react-native-localization';
import dayjs from 'dayjs';
import relativeTime from 'dayjs/plugin/relativeTime';
import 'dayjs/locale/en';
import 'dayjs/locale/fr';

import frStrings from '@assets/locales/fr';
import enStrings from '@assets/locales/en';

const strings = new LocalizedStrings({
  fr: frStrings,
  en: enStrings,
});

dayjs.locale(strings.getLanguage());
dayjs.extend(relativeTime);

export const updateAppLocale = (locale) => {
  dayjs.locale(locale);
  strings.setLanguage(locale);
};

export default strings;
