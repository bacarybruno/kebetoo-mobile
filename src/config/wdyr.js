import * as React from 'react';

if (process.env.NODE_ENV === 'development') {
  // eslint-disable-next-line import/no-extraneous-dependencies
  const whyDidYouRender = require('@welldone-software/why-did-you-render');
  const ReactRedux = require('react-redux');
  whyDidYouRender(React, {
    runtime: 'automatic',
    trackAllPureComponents: true,
    trackExtraHooks: [
      [ReactRedux, 'useSelector'],
    ],
  });
}
