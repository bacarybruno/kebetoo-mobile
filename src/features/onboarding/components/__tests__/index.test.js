import setupTest from '@app/config/jest-setup'
import { strings } from '@app/config'

import OnboardingSlide from '../index'

const givenOnboardingSlide = setupTest(OnboardingSlide)()

it('renders OnboardingSlide', () => {
  const { wrapper } = givenOnboardingSlide({
    slideTitle: strings.onboarding.screen_one_title,
    slideDescription: strings.onboarding.screen_one_description,
  })
  expect(wrapper.toJSON()).toMatchSnapshot()
})
