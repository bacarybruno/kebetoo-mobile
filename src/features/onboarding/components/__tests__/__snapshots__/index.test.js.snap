// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders OnboardingSlide 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "zIndex": 1,
        },
        false,
      ]
    }
  >
    <View
      accessibilityElementsHidden={false}
      importantForAccessibility="auto"
      onLayout={[Function]}
      pointerEvents="box-none"
      style={null}
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgb(255, 255, 255)",
              "borderBottomColor": "rgb(216, 216, 216)",
              "flex": 1,
              "shadowColor": "rgb(216, 216, 216)",
              "shadowOffset": Object {
                "height": 0.5,
                "width": 0,
              },
              "shadowOpacity": 0.85,
              "shadowRadius": 0,
            }
          }
        />
      </View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "maxHeight": undefined,
            "minHeight": undefined,
            "opacity": undefined,
            "transform": undefined,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "height": 0,
            }
          }
        />
        <View
          pointerEvents="box-none"
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "marginHorizontal": 16,
                "opacity": 1,
              }
            }
          >
            <Text
              accessibilityRole="header"
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 17,
                  "fontWeight": "600",
                }
              }
            >
              jest
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "opacity": 1,
          }
        }
      />
      <View
        accessibilityElementsHidden={false}
        closing={false}
        gestureVelocityImpact={0.3}
        importantForAccessibility="auto"
        onClose={[Function]}
        onGestureBegin={[Function]}
        onGestureCanceled={[Function]}
        onGestureEnd={[Function]}
        onOpen={[Function]}
        onTransitionStart={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        transitionSpec={
          Object {
            "close": Object {
              "animation": "spring",
              "config": Object {
                "damping": 500,
                "mass": 3,
                "overshootClamping": true,
                "restDisplacementThreshold": 10,
                "restSpeedThreshold": 10,
                "stiffness": 1000,
              },
            },
            "open": Object {
              "animation": "spring",
              "config": Object {
                "damping": 500,
                "mass": 3,
                "overshootClamping": true,
                "restDisplacementThreshold": 10,
                "restSpeedThreshold": 10,
                "stiffness": 1000,
              },
            },
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            forwardedRef={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            style={
              Object {
                "flex": 1,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "rgb(242, 242, 242)",
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": -1,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 5,
                  "top": 0,
                  "width": 3,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                    },
                    undefined,
                  ],
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column-reverse",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "marginHorizontal": 16,
                        "marginTop": 733.7,
                      }
                    }
                  >
                    <Text
                      adjustsFontSizeToFit={true}
                      bold={true}
                      numberOfLines={2}
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "color": "#000000",
                            "fontFamily": "System",
                            "fontSize": 28,
                            "fontWeight": "600",
                            "letterSpacing": 0.355469,
                            "lineHeight": 34,
                          },
                          Array [
                            Object {
                              "color": "rgba(0, 0, 0, 255)",
                            },
                            undefined,
                            undefined,
                          ],
                        ]
                      }
                    >
                      Join Our Social Media
                    </Text>
                    <View
                      style={
                        Object {
                          "marginTop": 46.690000000000005,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "opacity": 0,
                          }
                        }
                      >
                        <View
                          onLayout={[Function]}
                        >
                          <Text
                            numberOfLines={5}
                            style={Object {}}
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "color": "#000000",
                                    "fontFamily": "System",
                                    "fontSize": 17,
                                    "fontWeight": "400",
                                    "letterSpacing": -0.408,
                                    "lineHeight": 22,
                                  },
                                  undefined,
                                  Object {
                                    "color": "rgba(0, 0, 0, 255)",
                                  },
                                ]
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Array [
                                      Object {
                                        "backgroundColor": "transparent",
                                        "color": "#000000",
                                        "fontFamily": "System",
                                        "fontSize": 17,
                                        "fontWeight": "400",
                                        "letterSpacing": -0.408,
                                        "lineHeight": 22,
                                      },
                                      undefined,
                                      Object {
                                        "color": "rgba(0, 0, 0, 255)",
                                      },
                                    ],
                                    undefined,
                                  ]
                                }
                              >
                                Kebetoo is a microblogging app with 
                              </Text>
                              <Text
                                style={
                                  Array [
                                    Array [
                                      Object {
                                        "backgroundColor": "transparent",
                                        "color": "#000000",
                                        "fontFamily": "System",
                                        "fontSize": 17,
                                        "fontWeight": "400",
                                        "letterSpacing": -0.408,
                                        "lineHeight": 22,
                                      },
                                      undefined,
                                      Object {
                                        "color": "rgba(0, 0, 0, 255)",
                                      },
                                    ],
                                    Object {
                                      "backgroundColor": "transparent",
                                      "color": "rgba(0, 0, 0, 255)",
                                      "fontFamily": "System",
                                      "fontSize": 17,
                                      "fontWeight": "600",
                                      "letterSpacing": -0.408,
                                      "lineHeight": 22,
                                    },
                                  ]
                                }
                              >
                                African DNA
                              </Text>
                              <Text
                                style={
                                  Array [
                                    Array [
                                      Object {
                                        "backgroundColor": "transparent",
                                        "color": "#000000",
                                        "fontFamily": "System",
                                        "fontSize": 17,
                                        "fontWeight": "400",
                                        "letterSpacing": -0.408,
                                        "lineHeight": 22,
                                      },
                                      undefined,
                                      Object {
                                        "color": "rgba(0, 0, 0, 255)",
                                      },
                                    ],
                                    undefined,
                                  ]
                                }
                              >
                                . Be one of the first to join the adventure 💪
                              </Text>
                            </Text>
                          </Text>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Object {
                              "left": 0,
                              "opacity": 0,
                              "position": "absolute",
                              "top": 0,
                            }
                          }
                        >
                          <Text
                            style={Object {}}
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "color": "#000000",
                                    "fontFamily": "System",
                                    "fontSize": 17,
                                    "fontWeight": "400",
                                    "letterSpacing": -0.408,
                                    "lineHeight": 22,
                                  },
                                  undefined,
                                  Object {
                                    "color": "rgba(0, 0, 0, 255)",
                                  },
                                ]
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Array [
                                      Object {
                                        "backgroundColor": "transparent",
                                        "color": "#000000",
                                        "fontFamily": "System",
                                        "fontSize": 17,
                                        "fontWeight": "400",
                                        "letterSpacing": -0.408,
                                        "lineHeight": 22,
                                      },
                                      undefined,
                                      Object {
                                        "color": "rgba(0, 0, 0, 255)",
                                      },
                                    ],
                                    undefined,
                                  ]
                                }
                              >
                                Kebetoo is a microblogging app with 
                              </Text>
                              <Text
                                style={
                                  Array [
                                    Array [
                                      Object {
                                        "backgroundColor": "transparent",
                                        "color": "#000000",
                                        "fontFamily": "System",
                                        "fontSize": 17,
                                        "fontWeight": "400",
                                        "letterSpacing": -0.408,
                                        "lineHeight": 22,
                                      },
                                      undefined,
                                      Object {
                                        "color": "rgba(0, 0, 0, 255)",
                                      },
                                    ],
                                    Object {
                                      "backgroundColor": "transparent",
                                      "color": "rgba(0, 0, 0, 255)",
                                      "fontFamily": "System",
                                      "fontSize": 17,
                                      "fontWeight": "600",
                                      "letterSpacing": -0.408,
                                      "lineHeight": 22,
                                    },
                                  ]
                                }
                              >
                                African DNA
                              </Text>
                              <Text
                                style={
                                  Array [
                                    Array [
                                      Object {
                                        "backgroundColor": "transparent",
                                        "color": "#000000",
                                        "fontFamily": "System",
                                        "fontSize": 17,
                                        "fontWeight": "400",
                                        "letterSpacing": -0.408,
                                        "lineHeight": 22,
                                      },
                                      undefined,
                                      Object {
                                        "color": "rgba(0, 0, 0, 255)",
                                      },
                                    ],
                                    undefined,
                                  ]
                                }
                              >
                                . Be one of the first to join the adventure 💪
                              </Text>
                            </Text>
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
